### DISCUSSION TITLE
Is there any method to check what are registered?

### DISCUSSION THREAD

**Discussion by hyungjunk** [react-hook-form/react-hook-form#7620] [Q&A] [answered]
If an input is dynamically registered, is there any method to check what are really registered ?

It seems `shouldUnregister: true` doesn't work in my case even though that input is not rendered.

I may have to check what are currently registered and validate only those, unless the form complain the input value is not valid although the input is not actually registered.

**Comment by bluebill1049** [comment]
`getValues` should tell you if the input is still registered or not

**Reply by danielhammerl** [reply]
@bluebill1049 
not really, getValues returns the defaultValues merged with the new values from the form, no way to check if a field is registered.

Heres an example: https://codesandbox.io/s/react-hook-form-js-forked-v8qlg?file=/src/App.js

Click on button console.logs the getValues method, and you see there are fields logged which arent registered.

You can see the registered fields in control.__fields, but like doc says you shouldnt use control. I would need this feature too so an api which simply reads control.__fields would be great, i would even implement this feature by my self if its ok

**Reply by Moshyfawn** [reply]
Consider looking into the #7643 discussion. It might shed some light on the RHF mindset about registering fields etc.

**Comment by ritikbanger** [comment, marked as answer]
You can check if your input is registered or not with the help of `getValues` method. getValues will not trigger re-renders or subscribe to input changes, if your input changes, go for `watch`.

**Reply by PrimulaX** [reply]
What if you have `defaultValues`? That’s not solution
